### ISSUE TITLE
sendmsg_nonblock with exception: false gobbling all errors

### ISSUE THREAD

**Issue by HoneyryderChuck** [oracle/truffleruby#2400]
Reproducible with:

```ruby
require "socket"

host = "fe80::1213:31ff:fe67:8b94"
ip = IPAddr.new(host)
udp  = UDPSocket.new(ip.family)

puts udp.sendmsg_nonblock("bla", 0, Socket.sockaddr_in(53, host), exception: false)
```

In CRuby, this raises ` No route to host - sendmsg(2) (Errno::EHOSTUNREACH)`, whereas in truffleruby, it return `:wait_writable`.

Bottom line is, connectivity errors shouldn't  be gobbled.

Observed this behaviour with 21.1.0

**Comment by eregon** [comment]
@HoneyryderChuck Trying that script (+ `require 'ipaddr'`) on CRuby I get:
```
$ ruby -v sendmsg.rb
ruby 2.7.3p183 (2021-04-05 revision 6847ee089d) [x86_64-linux]
3
```
And the same on truffleruby-dev.

**Comment by HoneyryderChuck** [comment]
Probably because your IPv6 connectivity works better (kudos to your ISP provider 😂 ).

Maybe try it out with some other unreachable IP?

**Comment by HoneyryderChuck** [comment]
You can set the following ip table rule:

```
ip route add unreachable 192.168.2.1
```

And use that IP. Also doable from within truffleruby's docker container.
